forage-android/src/main/java/com/joinforage/forage/android/collect/BTPinCollector.kt

```diff
@@ -0,0 +1,100 @@
+package com.joinforage.forage.android.collect
+
+import com.basistheory.android.service.BasisTheoryElements
+import com.basistheory.android.service.ProxyRequest
+import com.joinforage.forage.android.BuildConfig
+import com.joinforage.forage.android.network.model.ForageApiError
+import com.joinforage.forage.android.network.model.ForageApiResponse
+import com.joinforage.forage.android.network.model.ForageError
+import com.joinforage.forage.android.ui.ForagePINEditText
+import org.json.JSONException
+
+internal class BTPinCollector(
+    private val pinForageEditText: ForagePINEditText,
+    private val merchantAccount: String
+) : PinCollector {
+    override suspend fun collectPinForBalanceCheck(
+        paymentMethodRef: String,
+        cardToken: String,
+        encryptionKey: String
+    ): ForageApiResponse<String> {
+        val bt = buildBt()
+
+        val proxyRequest: ProxyRequest = ProxyRequest().apply {
+            headers = mapOf(
+                "X-KEY" to encryptionKey,
+                "Merchant-Account" to merchantAccount,
+                "BT-PROXY-KEY" to PROXY_ID,
+                "Content-Type" to "application/json"
+            )
+            body = object {
+                val pin = pinForageEditText.getTextElement()
+                val card_number_token = cardToken
+            }
+            path = balancePath(paymentMethodRef)
+        }
+
+        val response = bt.proxy.post(proxyRequest)
```

Comment:
This part can also throw exceptions; all network errors will probably be thrown here.

That is the main purpose of running it inside the `runCatching`. You can also replace `runCatching` for a regular try-catch too

Reply:
Thank you for the insight! I was able to force BT to throw an error and test that my changes handled that response gracefully in this commit: [ccef382](https://github.com/teamforage/forage-android-sdk/pull/38/commits/ccef382e81787c87e7988bb9860170ccc064e31b).

Reply:
anytime =]